param.go

```diff
@@ -144,6 +147,14 @@ type paramList struct {
 	Params []param
 }
 
+func (pl paramList) GraphNode() []graphNode {
+	types := []graphNode{}
```

Comment:
nit: we can instantiate this as `var types []graphNode` to avoid an unnecessary allocation if `len(pl.Params) == 0`.